src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -205,11 +206,12 @@ private ComponentState AttachAndInitComponent(IComponent component, int parentCo
         /// </summary>
         /// <param name="eventHandlerId">The <see cref="RenderTreeFrame.AttributeEventHandlerId"/> value from the original event attribute.</param>
         /// <param name="eventArgs">Arguments to be passed to the event handler.</param>
+        /// <param name="fieldInfo">Information that the renderer can use to update the state of the existing render tree to match the UI.</param>
         /// <returns>
         /// A <see cref="Task"/> which will complete once all asynchronous processing related to the event
         /// has completed.
         /// </returns>
-        public virtual Task DispatchEventAsync(int eventHandlerId, UIEventArgs eventArgs)
+        public virtual Task DispatchEventAsync(int eventHandlerId, EventFieldInfo fieldInfo, UIEventArgs eventArgs)
```

Comment:
Is the expectation that this chain grows bigger than 1 when there are rapidly firing events?

Reply:
The chain of replaced event handler IDs? If so, yes, it can grow beyond 1.